### PR TITLE
Notebook tasks & support for arbitrary decorator tasks

### PR DESCRIPTION
**Pull request by kumare3**
- This PR makes it possible to create arbitrary decorator tasks and as
   an example creates a notebook task that uses papermill to execute
a notebook as a standalone task and captures its outputs.

 - Any task can be a notebook task

### CODE REVIEW

File: flytekit/taskplugins/notebook/task.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    One of the cells (usually the first) should be marked as the parameters cell. This task will inject inputs after this
+    cell. The task will inject the outputs observed from Flyte
+
+    Property 2:
+    For a notebook that produces outputs, that should be consumed by a subsequent notebook, use the method
+    :py:func:`record_outputs` in your notebook after the outputs are ready and pass all outputs.
+
+    Usage:
+
+    .. code-block:: pthon
```
Comment (katrogan) [inline flytekit/taskplugins/notebook/task.py:43]: python?

File: flytekit/taskplugins/notebook/task.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        notebook_path: str,
+        task_config: T = None,
+        inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        outputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        **kwargs,
+    ):
+        plugin_class = TaskPlugins.find_pythontask_plugin(type(task_config))
+        self._plugin = plugin_class(task_config=task_config, task_function=_dummy_task_func)
+        task_type = f"nb-{self._plugin.task_type}"
+        self._notebook_path = os.path.abspath(notebook_path)
```
Comment (katrogan) [inline flytekit/taskplugins/notebook/task.py:108]: oh interesting, does this assume notebook tasks are only initialized in container?

File: tests/flytekit/unit/taskplugins/notebook/test_task.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from flytekit import kwtypes
+from flytekit.taskplugins.notebook import NotebookTask
+
+
+def test_notebook_task():
+    nb = NotebookTask(
+        name="test", notebook_path="./test-notebook.ipynb", inputs=kwtypes(pi=float), outputs=kwtypes(square=float)
+    )
+    print(nb.execute(pi=4))
```
Comment (katrogan) [inline tests/flytekit/unit/taskplugins/notebook/test_task.py:9]: can you validate the outputs here too?

**Review by katrogan — APPROVED** [review]
super slick!

File: flytekit/taskplugins/notebook/task.py (hunk 2 above)
Comment (kumare3) [reply flytekit/taskplugins/notebook/task.py:120]: ya for the moment - they need the container

File: tests/flytekit/unit/taskplugins/notebook/test_task.py (hunk 3 above)
Comment (kumare3) [reply tests/flytekit/unit/taskplugins/notebook/test_task.py]: ohh i did it, updated the code

**Pull request merged**